src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonBulkCopyTest.cs

```diff
@@ -232,43 +237,46 @@ private async Task BulkCopyDataAsync(CommandBehavior cb, bool enableStraming)
                     long countEnd = System.Convert.ToInt32(await commandRowCount.ExecuteScalarAsync());
                     _output.WriteLine("Ending row count = {0}", countEnd);
                     _output.WriteLine("{0} rows were added.", countEnd - countStart);
+                    Assert.Equal(expectedTransferCount, countEnd - countStart);
                 }
             }
         }
 
+        [Theory]
         [InlineData(CommandBehavior.Default, false)]
         [InlineData(CommandBehavior.Default, true)]
         [InlineData(CommandBehavior.SequentialAccess, false)]
         [InlineData(CommandBehavior.SequentialAccess, true)]
-        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.IsJsonSupported))]
         public void TestJsonBulkCopy(CommandBehavior cb, bool enableStraming)
         {
-            PopulateData(10000);
+            int jsonArrayElements = 300;
+            int rows = 100;
+            PopulateData(jsonArrayElements, rows);
             using (SqlConnection connection = new SqlConnection(DataTestUtility.TCPConnectionString))
             {
-                BulkCopyData(cb, enableStraming);
+                BulkCopyData(cb, enableStraming, rows);
                 connection.Open();
-                PrintJsonDataToFile(connection);
-                CompareJsonFiles();
+                PrintJsonDataToFileAndCompare(connection);
                 DeleteFile(_jsonFile);
                 DeleteFile(_outputFile);
             }
         }
 
+        [Theory]
         [InlineData(CommandBehavior.Default, false)]
         [InlineData(CommandBehavior.Default, true)]
         [InlineData(CommandBehavior.SequentialAccess, false)]
         [InlineData(CommandBehavior.SequentialAccess, true)]
-        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.IsJsonSupported))]
         public async Task TestJsonBulkCopyAsync(CommandBehavior cb, bool enableStraming)
         {
-            PopulateData(10000);
+            int jsonArrayElements = 10;
+            int rows = 10000;
```

Comment:
Can you pass these integers as params to both `TestJsonBulkCopy` and `TestJsonBulkCopyAsync` and test both sync/async variants with these values?